### PR TITLE
Feature/accountsettings

### PR DESCRIPTION
**Pull request by chris7neves**
Added login authentication, account settings page (need to implement record update before finishing it completely)

### CODE REVIEW

File: public/account_settings.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+<?php }else{ ?>
+
+    <h5>You need to be logged in to see your account details.</h1>
+
+<?php } ?>
+
+
+<form method="post">
+
```
Comment (jamestjw) [inline public/account_settings.php:67]: This line is extraneous I think?

File: common.php [hunk 2]
```diff
@@ -32,3 +34,28 @@ function startsWith($string, $startString) {
...
+	if(isset($_SESSION)){
+	  $fname = $_SESSION["current_user"];
+	  return $fname->first_name;
+	}else{
+	  return '';
+	}
+  }
+
+  function auth_loggedin() {
+    if(isset($_SESSION) && array_key_exists("AuthKey", $_SESSION)){ // Session is started and authentication key exists
```
Comment (jamestjw) [inline common.php:49]: I feel like the name of the function is not a good hint of what the function does 🤔 I'm not too sure what I would name it, or it seems like the function plays the role of verifying that the user is logged in, idk maybe something like `verify_logged_in` or `verify_authentication` 😂 maybe you'll have more ideas

File: public/discussion.php [hunk 3]
```diff
@@ -3,6 +3,7 @@
 require_once "../common.php";
 
 ensure_logged_in();
+auth_loggedin();
```
Comment (jamestjw) [inline public/discussion.php:7]: Maybe the `auth_logged_in` method should be invoked in the `ensure_logged_in` function, what do you think

File: public/index.php [hunk 4]
```diff
@@ -1,5 +1,10 @@
 <?php include "templates/header.php"; ?>
 
+<?php 
+auth_loggedin();
+
+?>
+
```
Comment (jamestjw) [inline public/index.php:7]: What's the idea of invoking this here?

File: common.php [hunk 5]
```diff
@@ -32,3 +34,28 @@ function startsWith($string, $startString) {
...
+      // Get the auth token from session
+      $authtoken = $_SESSION["AuthKey"];
+      if(!Loggedin::find_by_user_digest($authtoken)){ // If user does not exist or authentication key does not match an entry in the loggedin table, logout
+        $_SESSION = array();
+        setcookie(session_name(), '', time() - 30000, '/');
+        session_destroy();
+      }
```
Comment (jamestjw) [inline common.php:56]: Should the user be redirected somewhere after the destruction of a session

Comment (jamestjw) [reply common.php:56]: btw what's the idea behind time()-30000?

**Review by jamestjw — CHANGES_REQUESTED** [review]

File: public/account_settings.php (hunk 1 above)
Comment (chris7neves) [reply public/account_settings.php:67]: Oh oops, youre right. Not sure why thats there

File: common.php (hunk 2 above)
Comment (chris7neves) [reply common.php:49]: Good idea, ill change it to verify_logged_in, i think that sums up what it does

File: public/discussion.php (hunk 3 above)
Comment (chris7neves) [reply public/discussion.php:7]: I was actually conflicted about that, i see ensure logged in as a method make sure that the user is logged in if they want to see a certain page. But verify_logged_in is more of a cleanup method to make sure that theres no mismatch between the users cookies and the database in a way

File: public/index.php (hunk 4 above)
Comment (chris7neves) [reply public/index.php:7]: We have a message in the index page right now that says "You are logged in". The method being there makes sure that the user is actually logged in and that "current_user" is not just being read from the _SESSION variable.

File: common.php (hunk 5 above)
Comment (chris7neves) [reply common.php:56]: No, I had a redirect there before, but they should only be redirected in ensure_loggedin. Some pages allow you to view them even if youre not logged in right?

The time() -30000 sets the cookie expire time to sometime in the past, so this terminates the cookie

File: public/discussion.php (hunk 3 above)
Comment (jamestjw) [reply public/discussion.php:7]: ok with that being the case, should `verify_logged_in` be in the header file then?

File: common.php (hunk 5 above)
Comment (jamestjw) [reply common.php:56]: ok in that case then we should have a comment there to briefly describe the logic behind `time() -30000`

File: public/discussion.php (hunk 3 above)
Comment (chris7neves) [reply public/discussion.php:7]: this is a good idea

**Review by jamestjw — APPROVED** [review]
man why are you introducing a ton of unnecessary white space into the code 😂 I installed this linter locally and it works pretty work, if you have a moment or two could you run it first before merging it https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/master/doc/installation.rst

**Pull request merged**